examples/inspection/plot_permutation_importance_multicollinear.py

```diff
@@ -18,68 +21,103 @@
 tree_indices = np.arange(0, len(clf.feature_importances_)) + 0.5
 
 fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 8))
-ax1.barh(tree_indices, clf.feature_importances_[tree_importance_sorted_idx], height=0.7)
-ax1.set_yticks(tree_indices)
-ax1.set_yticklabels(data.feature_names[tree_importance_sorted_idx])
-ax1.set_ylim((0, len(clf.feature_importances_)))
-ax2.boxplot(
-    result.importances[perm_sorted_idx].T,
-    vert=False,
-    labels=data.feature_names[perm_sorted_idx],
+mdi_importances.sort_values().plot.barh(ax=ax1)
+ax1.set_xlabel("Gini importance")
+plot_permutation_importance(clf, X_train, y_train, ax2)
+ax2.set_xlabel("Decrease in accuracy score")
+fig.suptitle(
+    "Impurity-based vs. permutation importances on multicollinear features (train set)"
 )
-fig.tight_layout()
-plt.show()
+_ = fig.tight_layout()
+
+# %%
+# The plot on the left shows the gini importance of the model. As the
+# scikit-learn implementation of
+# :class:`~sklearn.ensemble.RandomForestClassifier` uses a random subsets of
+# :math:`\sqrt{n_\text{features}}` features at each split, it is able to dilute
+# the dominance of any single correlated feature. As a result, the individual
+# feature importance may be distributed more evenly among the correlated
+# features. Since the features have large cardenality and the classifier is
+# non-overfitted, we can relatively trust those values.
+#
+# The permutation importance on the right plot shows that permuting a feature
+# drops the accuracy by at most `0.012`, which would suggest that none of the
+# features are important. This is in contradiction with the high test accuracy
+# computed as baseline: some feature must be important.
+#
+# Similarly, the change in accuracy score computed on the test set appears to be
+# driven by chance:
+
+fig, ax = plt.subplots(figsize=(7, 6))
+plot_permutation_importance(clf, X_test, y_test, ax)
+ax.set_title("Permutation Importances on multicollinear features\n(test set)")
+ax.set_xlabel("Decrease in accuracy score")
+_ = ax.figure.tight_layout()
 
 # %%
+# Nevertheless, one can still compute a meaningful permutation importance in the
+# presence of correlated features, as demonstrated in the following section.
+#
 # Handling Multicollinear Features
 # --------------------------------
-# When features are collinear, permutating one feature will have little
-# effect on the models performance because it can get the same information
-# from a correlated feature. One way to handle multicollinear features is by
-# performing hierarchical clustering on the Spearman rank-order correlations,
-# picking a threshold, and keeping a single feature from each cluster. First,
-# we plot a heatmap of the correlated features:
+# When features are collinear, permutating one feature will have little effect
+# on the models performance because it can get the same information from a
+# correlated feature. One way to handle multicollinear features is by performing
```

Comment:
```
When features are collinear, permuting one feature will have little effect
on the models performance because it can get the same information from a
correlated feature. Note that this is not the case for all predictive models
and depends on their underlying implementation.

One way to handle ...
```